### PR TITLE
Subscription improvements

### PR DESCRIPTION
**Pull request by jdalsem**
- [x] tests?

### CODE REVIEW

File: engine/classes/ElggUser.php [hunk 1]
```diff
@@ -40,6 +41,10 @@ protected function initializeAttributes() {
 		parent::initializeAttributes();
 		$this->attributes['subtype'] = 'user';
 		
+		$this->access_id = ACCESS_PUBLIC;
+		$this->owner_guid = 0; // Users aren't owned by anyone, even if they are admin created.
+		$this->container_guid = 0; // Users aren't contained by anyone, even if they are admin created.
```
Comment (jeabakker) [inline engine/classes/ElggUser.php:46]: needs to be set by `$this->attributes['<name>'] = <value>;`

File: engine/lib/sessions.php [hunk 2]
```diff
@@ -286,6 +286,7 @@ function login(\ElggUser $user, $persistent = false) {
 	
 	if ($first_login) {
 		elgg_trigger_event('login:first', 'user', $user);
+		$user->first_login = time();
```
Comment (jeabakker) [inline engine/lib/sessions.php:289]: maybe use `first_login_ts` and document the property with `ElggUser`

**Review by jeabakker — CHANGES_REQUESTED** [review]
also don't forget to change the admin user create action to generate a validated user

File: engine/lib/sessions.php (hunk 2 above)
Comment (jdalsem) [reply engine/lib/sessions.php:289]: not going to change... keeping name in line with other login metadata

**Review by jeabakker — APPROVED** [review]

**Pull request merged**